ale_linters/c/ccls.vim

```diff
@@ -0,0 +1,32 @@
+" Author: Ye Jingchen <ye.jingchen@gmail.com>, Ben Falconer <ben@falconers.me.uk>, jtalowell <jtalowell@protonmail.com>
+" Description: A language server for C
+
+call ale#Set('c_ccls_executable', 'ccls')
+call ale#Set('c_ccls_init_options', {})
+
+function! ale_linters#c#ccls#GetProjectRoot(buffer) abort
+    let l:project_root = ale#path#FindNearestFile(a:buffer, '.ccls-root')
+
+    if empty(l:project_root)
+        let l:project_root = ale#path#FindNearestFile(a:buffer, 'compile_commands.json')
+    endif
```

Comment:
This will do for now. Later on, I'll add a function for finding the project root based on `compile_commands.json` being inside of a directory like `build`.